msteams-platform/toolkit/use-CICD-template.md

```diff
@@ -10,233 +10,357 @@ ms.date: 04/20/2022
+              path: appPackage/build/appPackage.zip
+    ```
+
+    > [!NOTE]
+    > The default pipeline triggers when push events occur on the main branch. You've the option to modify it to suit your specific requirements.
+
+1. Go to GitHub.
+
+1. Update the following variables and secrets you created during the Prerequisites:
+
+    * `AZURE_SERVICE_PRINCIPAL_CLIENT_ID`, `AZURE_TENANT_ID`, and `AZURE_SERVICE_PRINCIPAL_CLIENT_SECRET`.
+
+      :::image type="content" source="../assets/images/teams-toolkit-v2/repo-settings.png" alt-text="Screenshot shows the repo settings.":::
+
+      > [!NOTE]
+      > The `AZURE_SERVICE_PRINCIPAL_CLIENT_SECRET` variable must be set as secret.
+      > Utilize the [GitHub environment](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#environment-variables) to use different sets of variables.
+
+    * Go to the `teamsapp.yml` file. In the `deploy` stage, the values enclosed in `${{}}` are the required variable keys. If you've used the `provision` command from Teams Toolkit, you can locate the values in the environment files in the `.env` folder.
+
+      :::image type="content" source="../assets/images/teams-toolkit-v2/teamsappyml.png" alt-text="Screenshot shows the bot Azure app service resource ID in teamsapp.yml file.":::
 
-1. Change the trigger: By default, the CI workflow is triggered when a new pull request is created against `dev` branch.
-1. Add scripts to build the project: By default, the `Build the project` step is commented.
-1. Add scripts to run unit test: By default, the `Run unit test` step is commented.
+    * Go to the `appPackage/manifest.json` file. The values enclosed in `${{}}` are the required variable keys. If you've used the `provision` command from Teams Toolkit, you can locate the values in the environment files in the `.env` folder.
 
-### Customize CD workflow
+      :::image type="content" source="../assets/images/teams-toolkit-v2/manifest.png" alt-text="Screenshot shows the Teams app ID in manifest file.":::
 
-To customize the CD workflow, you can do the following:
+1. In the GitHub, navigate to your repository’s **Settings** and select **Secrets and variables** > **Actions**.
+
+    Update the variable keys that you've gathered for the following variables:
 
-1. Change the trigger: By default, the CD workflow is triggered when new commits are pushed into `main` branch.
-1. Change the value of environment variable `TEAMSFX_ENV_NAME`: By default, the value is `dev`.
-1. Change the value of environment variable `TEAMSFX_CLI_VERSION`: By default, the value is `2.*`.
-1. Add scripts to build the project: By default, the `Build the project` step is commented.
-1. Add scripts to run unit test: By default, the `Run unit test` step is commented.
+    * `AZURE_SERVICE_PRINCIPAL_CLIENT_ID`.
+    * `AZURE_TENANT_ID`.
+    * `AZURE_SERVICE_PRINCIPAL_CLIENT_SECRET`.
+    * `BOT_AZURE_APP_SERVICE_RESOURCE_ID`
```

Comment:
Do you think there's a good way to express that these few variables are just an example? To prevent users to think that these variables are needed for all project templates.